app/src/lib/stores/app-store.ts

```diff
@@ -2922,9 +2926,10 @@ export class AppStore extends TypedBaseStore<IAppState> {
       const isBackgroundTask = true
       const gitStore = this.gitStoreCache.get(repo)
 
+      const progressCallback = repo.codespace? undefined : (progress: IFetchProgress) => { this.updatePushPullFetchProgress(repo, progress) }
```

Comment:
I'm very curious what we'll want to do about handling the progress callbacks.

Options I see:

1. Ignore progress callbacks for remote execution (what we're currently doing). If we want to mainline this I'm not sure that's the right approach. It would be hard to write code in a way that you have to remember that your callback might be completely ignored in some cases 😅
2. Support callbacks for remote execution. I don't think we'd want to try and serialize functions to send on the wire, but we could move from a message-based (currently http) to a stream-based transport and make the remote execution feel _much_ more like a subprocess where we have channels for stdin/stdout/stderr we can read/write to like they're file descriptors.